include/hipSYCL/sycl/libkernel/accessor.hpp

```diff
@@ -111,6 +111,18 @@ constexpr accessor_variant deduce_accessor_variant(mode_tag<TagDescT> tag,
   }
 }
 
+#if defined(_WIN32) && defined(__clang__) && defined(__CUDA__)
+#define WARN_ATTR(condition, text) __attribute__((diagnose_if(condition, text, "warning")))
+#else
+#define WARN_ATTR(condition, text)
+#endif
+
+constexpr bool warn_if_variant_on_windows(accessor_variant V) 
+WARN_ATTR(V != accessor_variant::true_t && V != accessor_variant::false_t,
```

Comment:
I think we could also allow `ranged_placeholder` as it should be just as general as `true_t` and `false_t` and not have any empty base classes.